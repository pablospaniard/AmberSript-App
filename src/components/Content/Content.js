import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Grid } from '@material-ui/core'
import { saveAs } from 'file-saver/FileSaver'

import { List, Item } from '../'
import { Spinner } from '../UI'

import styles from './Content.scss'

class Content extends Component {
  state = {
    step: 0
  }

  onDecodeButtonHandler = () => {
    this.setState({
      step: 1
    })
  }

  onBackButtonHandler = () => {
    this.setState({
      step: 0
    })
  }

  onSaveJSONHandler = () => {
    const text = document.getElementById('save-text').innerText
    var blob = new Blob([text], { type: 'text/plain;charset=utf-8' })
    saveAs(blob, 'AmberScript_your_text.txt')
  }

  render() {
    const { step } = this.state
    const { data, error, loading, src } = this.props
    let content
    switch (step) {
      case 0:
        content = loading ? (
          <Spinner />
        ) : (
          <List data={data} onDecodeClick={this.onDecodeButtonHandler} />
        )
        break
      case 1:
        content = loading ? (
          <Spinner />
        ) : (
          <Item
            src={src}
            onBackClick={this.onBackButtonHandler}
            onSaveClick={this.onSaveJSONHandler}
            data={data}
          />
        )
        break
      default:
        break
    }

    if (error) {
      content = <p> Something went wrong, please try again</p>
    }

    return (
      <Grid container className={styles.Content}>
        {content}
      </Grid>
    )
  }
}

Content.propTypes = {
  data: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  error: PropTypes.object,
  src: PropTypes.string,
  loading: PropTypes.bool
}

export default Content
