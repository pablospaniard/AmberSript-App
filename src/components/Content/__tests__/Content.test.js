import React from 'react'
import { render, cleanup } from 'react-testing-library'
import Content from '../Content'

describe('Content should render properly', () => {
  afterEach(cleanup)

  const data = {
    recordId: 'test',
    segments: [
      {
        words: [
          {
            text: 'Hi',
            start: 0.2,
            end: 0.8
          },
          {
            text: 'welcome',
            start: 0.8,
            end: 1.5
          },
          {
            text: 'at',
            start: 1.5,
            end: 1.8
          },
          {
            text: 'Amber',
            start: 1.8,
            end: 2.1
          },
          {
            text: 'script.',
            start: 2.1,
            end: 2.6
          }
        ],
        speaker: 'Amber'
      }
    ]
  }

  it('should match snapshot', () => {
    const { container } = render(<Content data={data} />)
    expect(container.firstChild).toMatchSnapshot('Content_snapshot_1')
  })

  it('should show Spinner if loading is true', () => {
    const { getByTestId } = render(<Content loading data={data} />)
    getByTestId('spinner')
  })

  it('should show List if loading is false', () => {
    const { getByText } = render(<Content data={data} />)
    getByText('Decode')
  })
})
