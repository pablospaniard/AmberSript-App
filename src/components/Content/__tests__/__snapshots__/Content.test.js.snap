// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render properly should match snapshot: Content_snapshot_1 1`] = `
<div
  class="MuiGrid-container-1 Content"
>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div
      class="MuiPaper-root-98 MuiPaper-elevation2-102 MuiPaper-rounded-99 Paper"
    >
      <h2
        class="MuiTypography-root-125 MuiTypography-title-131"
      >
        test
      </h2>
      <div
        class="Buttons"
      >
        <button
          class="Button_Select null"
          type="button"
        >
          Decode
        </button>
        <button
          class="null Button_Delete"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
